vmdb/app/helpers/provider_foreman_helper.rb

```diff
@@ -1,30 +1,126 @@
+    h[:value] = @record.hostname
+    h
   end
 
-  def textual_configuration_profile
-    {:label => "Configuration Profile", :value => @record.configuration_profile.try(:name)}
+  def textual_ipmi_present
+    {:label => "IPMI Present", :value => @record.ipmi_present}
   end
 
-  def textual_configuration_profile_desc
-    {:label => "Configuration Profile Description", :value => @record.configuration_profile.try(:description)}
+  def textual_ipaddress
+    {:label => "IP Address", :value => @record.ipaddress}
   end
 
-  def textual_provider
-    {:label => "Foreman Provider", :value => @record.configuration_manager.provider.name}
+  def textual_mac_address
+    {:label => "Mac Address", :value => @record.mac_address}
   end
 
-  def textual_provider_url
-    {:label => "Foreman Provider URL", :value => @record.configuration_manager.provider.url}
+  def textual_configuration_profile_desc
+    h = {}
+    h[:label] = "Configuration Profile Description"
+    h[:image] = "configuration_profile"
+    h[:value] = @record.configuration_profile.try(:description)
+    h[:explorer] = true
+    h
+  end
+
+  def textual_provider_name
+    h = {}
+    h[:label] = "Provider"
+    h[:image] = "provider_foreman"
+    h[:value] = @record.configuration_manager.try(:name)
+    h[:explorer] = true
+    h
   end
 
   def textual_zone
     {:label => "Zone", :value => @record.configuration_manager.my_zone}
   end
+
+  def textual_group_tags
+    items = %w(tags)
+    items.collect { |m| send("textual_#{m}") }.flatten.compact
+  end
+
+  def textual_tags
+    label = "#{session[:customer_name]} Tags"
+    h = {:label => label}
+    tags = session[:assigned_filters]
+    if tags.empty?
+      h[:image] = "smarttag"
+      h[:value] = "No #{label} have been assigned"
+    else
+      h[:value] =
+        tags.sort_by { |category, _assigned| category.downcase }.collect { |category, assigned| {:image => "smarttag", :label => category, :value => assigned} }
+    end
+    h
+  end
+
+  def textual_group_environment
+    items = %w(configuration_environment_name configuration_domain_name configuration_realm_name)
+    items.collect { |m| send("textual_#{m}") }.flatten.compact
+  end
+
+  def textual_configuration_environment_name
+    {:label => "Environment", :value => @record.configuration_profile.try(:configuration_environment_name)}
+  end
+
+  def textual_configuration_domain_name
+    {:label => "Domain", :value => @record.configuration_profile.try(:configuration_domain_name)}
+  end
+
+  def textual_configuration_realm_name
+    {:label => "Realm", :value => @record.configuration_profile.try(:configuration_realm_name)}
+  end
+
+  def textual_group_os
+    items = %w(configuration_compute_profile_name
+               configuration_architecture_name
+               operating_system_flavor_name
+               customization_script_medium_name
+               customization_script_ptable_name)
+    items.collect { |m| send("textual_#{m}") }.flatten.compact
```

Comment:
> ```
[textual_configuration_compute_profile_name, textual_configuration_architecture_name,
textual_operating_system_flavor_name, textual_customization_script_medium_name,
textual_customization_script_ptable_name].flatten.compact
```
Took your suggestion. Got rid of the iterator and ```send``` calls.
Looks much cleaner. Thanks.